systemtest/src/test/java/io/strimzi/systemtest/security/SecurityST.java

```diff
@@ -1328,6 +1335,148 @@ void testOwnerReferenceOfCASecrets() {
         });
     }
 
+    X509Certificate getCertificateFromSecret(Secret secret, String dataKey) {
+        String caCert = secret.getData().get(dataKey);
+        byte[] decoded = Base64.getDecoder().decode(caCert);
+        X509Certificate cacert = null;
+        try {
+            cacert = (X509Certificate)
+                    CertificateFactory.getInstance("X.509").generateCertificate(new ByteArrayInputStream(decoded));
+        } catch (CertificateException e) {
+            e.printStackTrace();
+        }
+        return cacert;
+    }
+
+    Secret getClusterSecret(String secretName) {
+        Secret secret = kubeClient().getSecret(CLUSTER_NAME_PREFIX + secretName, false);
+        if (secret == null) {
+            LOGGER.info("Systemwide secret not found. Using Cluster CAs validity {}",
+                    clusterName + "-" + CLUSTER_NAME_PREFIX + secretName);
+            secret = kubeClient().getSecret(clusterName + "-" + secretName, true);
+        }
+        LOGGER.debug("Using secret: {}", secret);
+        return secret;
+    }
+
+    @Test
+    void testClusterCACertRenew() {
+        KafkaResource.createAndWaitForReadiness(KafkaResource.kafkaEphemeral(clusterName, 3)
+            .editOrNewSpec()
+                .withNewClusterCa()
+                    .withRenewalDays(15)
+                    .withValidityDays(20)
+                .endClusterCa()
+            .endSpec()
+            .build());
+
+        KafkaUtils.waitForKafkaReady(clusterName);
+
+        Secret clientsCASecret = getClusterSecret("cluster-ca-cert");
+        X509Certificate cacert = getCertificateFromSecret(clientsCASecret, "ca.crt");
+        Date initialCertStartTime = cacert.getNotBefore();
+        Date initialCertEndTime = cacert.getNotAfter();
+
+        String dateBrokerCertCreation = getClusterSecret("kafka-brokers").getMetadata().getManagedFields().get(0).getTime();
+        ZonedDateTime initialBrokerCertCreationDate = ZonedDateTime.parse(dateBrokerCertCreation);
+
+        LOGGER.info("Change of kafka validity and renewal days - reconciliation should start.");
+        CertificateAuthority newClusterCA = new CertificateAuthority();
+        newClusterCA.setRenewalDays(150);
+        newClusterCA.setValidityDays(200);
+        KafkaResource.replaceKafkaResource(clusterName, k -> k.getSpec().setClusterCa(newClusterCA));
+
+        // Wait for reconciliation and verify certs have been updated
+        Map<String, String> kafkaPods = StatefulSetUtils.ssSnapshot(KafkaResources.kafkaStatefulSetName(clusterName));
+        StatefulSetUtils.waitTillSsHasRolled(KafkaResources.kafkaStatefulSetName(clusterName), 3, kafkaPods);
+        KafkaUtils.waitForKafkaReady(clusterName);
```

Comment:
You should pick snapshot before replacement of the resource, now the snapshot of the pods can be changed, due to rolling update. So this can end up with failure, because the `waitTillSsHasRolled` will wait when all snapshots will be changed

Reply:
fixuped